mlaut/benchmarking/evaluation.py

```diff
@@ -0,0 +1,525 @@
+
+        wilcoxon_df = pd.DataFrame()
+        values = np.array([])
+        prod = itertools.product(metrics_per_estimator_dataset.keys(), repeat=2)
+        for p in prod:
+            estim_1 = p[0]
+            estim_2 = p[1]
+            w, p_val = stats.wilcoxon(metrics_per_estimator_dataset[p[0]],
+                                      metrics_per_estimator_dataset[p[1]])
+
+            w_test = {
+                "estimator_1": estim_1,
+                "estimator_2": estim_2,
+                "statistic": w,
+                "p_val": p_val
+            }
+
+            wilcoxon_df = wilcoxon_df.append(w_test, ignore_index=True)
+            values = np.append(values, w)
+            values = np.append(values, p_val)
+
+        index = wilcoxon_df["estimator_1"].unique()
+        values_names = ["statistic", "p_val"]
+        col_idx = pd.MultiIndex.from_product([index, values_names])
+        values_reshaped = values.reshape(len(index), len(values_names) * len(index))
+
+        values_df_multiindex = pd.DataFrame(values_reshaped, index=index, columns=col_idx)
+
+        return wilcoxon_df, values_df_multiindex
+
+    def friedman_test(self, metric_name=None):
+        """
+        The Friedman test is a non-parametric statistical test used to detect differences
+        in treatments across multiple test attempts. The procedure involves ranking each row (or block) together,
+        then considering the values of ranks by columns.
+        Implementation used:
+        `scipy.stats <https://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.stats.friedmanchisquare.html>`_.
+        """
+        self._check_is_evaluated()
+        metric_name = self._validate_metric_name(metric_name)
+        metrics_per_estimator_dataset = self._get_metrics_per_estimator_dataset(metric_name)
+
+        friedman_test = stats.friedmanchisquare(
+            *[metrics_per_estimator_dataset[k] for k in metrics_per_estimator_dataset.keys()])
+        values = [friedman_test[0], friedman_test[1]]
+        values_df = pd.DataFrame([values], columns=["statistic", "p_value"])
+
+        return friedman_test, values_df
+
+    def nemenyi(self, metric_name=None):
+        """
+        Post-hoc test run if the `friedman_test` reveals statistical significance.
+        For more information see `Nemenyi test <https://en.wikipedia.org/wiki/Nemenyi_test>`_.
+        Implementation used `scikit-posthocs <https://github.com/maximtrp/scikit-posthocs>`_.
+        """
+        self._check_is_evaluated()
+        metric_name = self._validate_metric_name(metric_name)
+        metrics_per_estimator_dataset = self._get_metrics_per_estimator_dataset(metric_name)
+
+        strategy_dict = pd.DataFrame(metrics_per_estimator_dataset)
+        strategy_dict = strategy_dict.melt(var_name="groups", value_name="values")
+        nemenyi = sp.posthoc_nemenyi(strategy_dict, val_col="values", group_col="groups")
```

Comment:
given that we use scikit-posthocs only for a single method at the moment, it may be worth considering making it a soft dependency or even copying the necessary code